### PR TITLE
Objective-C compatibility

### PR DESCRIPTION
**Pull request by felipowsky**
- Objective-C compatibility
- Optional delegate functions

### CODE REVIEW

**Comment by joshwalker** [comment]
It would be better to implement a protocol extension that has implementations of the delegate methods that don't do anything.  That way you can override them if you want.  Lets keep the @objc out of our Swift code.  Unfortunately there isn't much you can do on the Enum.

File: HidingNavigationBar/HidingNavigationBarManager.swift [hunk 1]
```diff
@@ -56,6 +56,9 @@ public class HidingNavigationBarManager: NSObject, UIScrollViewDelegate, UIGestu
 		if viewController.navigationController == nil || viewController.navigationController?.navigationBar == nil {
 			fatalError("ViewController must be within a UINavigationController")
 		}
+        
+        viewController.extendedLayoutIncludesOpaqueBars = true
```
Comment (joshwalker) [inline HidingNavigationBar/HidingNavigationBarManager.swift]: This is a separate issue from ObjC compatibility, and should be split into it's own PR/issue.

**Comment by tristanhimmelman** [comment]
I agree with the comments @joshwalker made above.

**Comment by felipowsky** [comment]
I've removed `extendedLayoutIncludesOpaqueBars` from this pull request.

I think using `optional` has the same effect of what @joshwalker suggested.
The use of `?` for optional functions on Swift 2 doesn't require extra implementation or checking. Also it is clearer whether a delegate function is optional or not which in my opinion results in cleaner code.

Is there any reason for not using `@objc` and for using `Enum:String` instead of a `Enum:Int`?

I will guess you don't want to limit Swift's capabilities with Objective-C compatibility and I understand. I just needed this for my Objective-C project.

**Comment by tristanhimmelman** [comment]
@felipowsky the reason I don't want to include `@objc` is simply because I prefer to keep this project swift only. Thanks for your PRs
